### DISCUSSION TITLE
How to add the authentication key to EKB      R35.5.0

### DISCUSSION THREAD

**Discussion by Red-Dragon-99** [OE4T/meta-tegra#1581] [Q&A]
Hi everyone!

I fused my Jetson Orin Devkit (only SBC and PKC) and I created PK/KEK/db keys for UEFI. This way I could enable Secure Boot and UEFI Secure Boot using the meta-tegra kirkstone branch. Using R35.4 this works like a charm.  Now, when I updated to R35.5.0, the boot process fails.

I learned from [this discussion](https://github.com/OE4T/meta-tegra/discussions/1529), that with R35.5.0 UEFI authentication is enabled per default.

So, currently I'm trying to generate & flash an EKS image with authentication key as described in this NVIDIA forum [post](https://forums.developer.nvidia.com/t/jetpack-5-1-3-boot-error-with-all-security-enabled/284400/8).
It boils down to generating the authentication key and then invoking gen_ekb.py with this key as parameter.

However, it looks like I missed something. This would only generate the authentication key and put it into the eks image file to be flashed to the ekb partition. But where is this authentication key used? In other words where is the key used to lock something (I guess the uefi variables?). I didn't find any hints where this key is used. Can you please point me to that?
How do I sign and encrypt the eks.img to flash it to the device? Can you please point me to the file/yocto recipe to do that?

Any help is appreciated.

Thanks!

**Comment by madisongh** [comment]
The use of this added key is mentioned in the L4T documentation [here](https://docs.nvidia.com/jetson/archives/r36.3/DeveloperGuide/SD/Security/SecureBoot.html#sd-security-secureboot-uefivariableprotection).

How to generate an EKB and include it in your builds is described on [this wiki page](https://github.com/OE4T/meta-tegra/wiki/Secure-Boot-Support-in-L4T-R35.2.1-and-later).

You should be able to update the EKB with a UEFI update capsule. I don't think we have a wiki page on update capsules, but you should find general info on them in the L4T documentation. The `tegra-uefi-capsules` recipe generates them in your Yocto builds.
